internal/provider/data_source_consul_cluster_test.go

```diff
@@ -0,0 +1,64 @@
+package provider
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+var testDataSourceConsulClusterConfig = `
+resource "hcp_hvn" "test" {
+	hvn_id         = "test-hvn"
+	cloud_provider = "aws"
+	region         = "us-west-2"
+}
+
+resource "hcp_consul_cluster" "test" {
+	cluster_id = "test-consul-cluster"
+	hvn_id     = hcp_hvn.test.hvn_id
+	tier       = "development"
+}
+
+data "hcp_consul_cluster" "test" {
```

Comment:
I'm getting a bit concerned about test time. Since we have to create the resource here, maybe we could take this config and add it in a new test step on the resource tests. We'll have to document the pattern/leave comments, but I think it's worthwhile to save test time, since the creation/teardown of these cloud resources takes at least 10 minutes.